utils/s2n_map.h

```diff
@@ -30,3 +31,9 @@ S2N_RESULT s2n_map_complete(struct s2n_map *map);
 S2N_RESULT s2n_map_unlock(struct s2n_map *map);
 S2N_RESULT s2n_map_lookup(const struct s2n_map *map, struct s2n_blob *key, struct s2n_blob *value, bool *key_found);
 S2N_RESULT s2n_map_free(struct s2n_map *map);
+S2N_RESULT s2n_map_size(struct s2n_map *map, uint32_t *size);
+
+S2N_RESULT s2n_map_iterator_new(struct s2n_map *map, struct s2n_map_iterator **iter);
+S2N_RESULT s2n_map_iterator_next(struct s2n_map_iterator *iter, struct s2n_blob *value);
+S2N_RESULT s2n_map_iterator_has_next(const struct s2n_map_iterator *iter, bool *has_next);
+int s2n_map_iterator_free(struct s2n_map_iterator *iter);
```

Comment:
This doesn't have to be int-- we added a specific type of return for this case :) https://github.com/aws/s2n-tls/blob/main/utils/s2n_result.h#L58-L61